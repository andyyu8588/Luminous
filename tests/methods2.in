class A {
	private a;
	protected b;
	public c;
	
	public constructor() {
		this.a = "A's a";
		this.b = "A's b";
		this.c = "A's c";
	}
	
	private printA() {
		print(this.a);
	}

	protected printB() {
		print(this.b);
        print("From class A");
	}

	public printC() {
		this.printA();
		print(this.c);
	}
}

class B inherits A {
    private d;
    public e;

    public constructor(d) {
        this.b = "B's b";
        this.c = "B's c";
        this.d = d;
    }

    public printD() {
        this.printB();
        super.printB();
        print(this.d);
    }

    private printB() {
        print(this.b);
        print("From class B");
    }
}

instanceA = A();
instanceB = B("d");
instanceB.printD();
instanceA.printC();