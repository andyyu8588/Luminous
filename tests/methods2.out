B's b
From class B
B's b
From class A
d
A's a
A's c
